File: src/main/java/legacy/kms/chapter01/TicketSeller.java
package legacy.kms.chapter01;

class TicketSeller {

    private final TicketOffice ticketOffice;

    public TicketSeller(TicketOffice ticketOffice) {
        this.ticketOffice = ticketOffice;
    }

    public void sellTo(Audience audience){
        Ticket ticket = ticketOffice.getTicket();
        long amount = audience.buy(ticket);
        ticketOffice.plusAmount(amount);
    }
}

File: src/main/kotlin/kms/chapter01/TicketSeller.kt
package kms.chapter01

internal class TicketSeller(
    private val ticketOffice: TicketOffice,
) {

    fun sellTo(audience: Audience) {
        val ticket = ticketOffice.getTicket()
        if (ticket != null) {
            val amount = audience.buy(ticket)
            ticketOffice.plusAmount(amount)
        }
    }
}
